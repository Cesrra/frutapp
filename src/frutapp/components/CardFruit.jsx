import { Card, CardActionArea, CardContent, CardMedia, Grid, Typography } from '@mui/material';

export const CardFruit = ({ item, custom }) => {
  
    const priceFormat = new Intl.NumberFormat('es-CO', {
      style: 'currency',
      currency: 'COP',
      minimumFractionDigits: 0,
      maximumFractionDigits: 0,
    }).format(item.price);
  
    return (
    <Card key={item.id}>
      <CardActionArea>
        <CardMedia
          component="img"
          style={{ objectFit: 'cover', width: '100%', height: custom.height }}
          src={ item.imgUrl }
          alt={ item.name }
        />
        <CardContent>
            <Grid container justifyContent="space-between" alignItems="center">
                <Grid item>
                    <Typography gutterBottom variant="h6" component="div">
                        { item.name }
                    </Typography>
                </Grid> 
                <Grid item>
                    <Typography gutterBottom variant="h7" component="div">
                        {`${priceFormat} cop/kg`} 
                    </Typography>
                </Grid>
          </Grid>
          <Typography variant="body2" color="text.secondary">
            <strong>{ `Alrededor de ${item.calories}` }</strong> 
            <hr />
            { item.description }
          </Typography>
        </CardContent>
      </CardActionArea>
    </Card>
  )
}
