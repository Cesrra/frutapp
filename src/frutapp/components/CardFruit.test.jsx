import { render } from "@testing-library/react"
import { CardFruit } from "./CardFruit"


const fruit = {
    id: 1,
    calories: '60 kcal por 100g',
    imgUrl: 'https://okdiario.com/img/vida-sana/2017/01/26/mango_verde_maduro.jpg',
    name: 'Mango',
    description: 'Mango tropical: joya dorada de dulzura veraniega. Pulpa jugosa y vibrante color en cada bocado, una experiencia exquisita. ðŸ¥­ðŸŒž',
    price: 2000,
}

describe("Pruebas en el componente <CardFruit />", () => {
    test('El componente de CardFruit debe hacer Match con el Snapshot', () => {
        const { container } = render(<CardFruit item={fruit} custom={{ height: '170px' }} />)
        expect(container).toMatchSnapshot()
    })
})