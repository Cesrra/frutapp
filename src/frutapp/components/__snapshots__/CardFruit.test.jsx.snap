// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <CardFruit /> El componente de CardFruit debe hacer Match con el Snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
  >
    <button
      class="MuiButtonBase-root MuiCardActionArea-root css-1oitmrd-MuiButtonBase-root-MuiCardActionArea-root"
      tabindex="0"
      type="button"
    >
      <img
        alt="Mango"
        class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
        src="https://okdiario.com/img/vida-sana/2017/01/26/mango_verde_maduro.jpg"
        style="object-fit: cover; width: 100%; height: 170px;"
      />
      <div
        class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container css-9cyib4-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <div
              class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-18k87ye-MuiTypography-root"
            >
              Mango
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <div
              class="MuiTypography-root MuiTypography-h7 MuiTypography-gutterBottom css-anmq6p-MuiTypography-root"
            >
              2000 cop/kg
            </div>
          </div>
        </div>
        <p
          class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
        >
          <strong>
            Alrededor de 60 kcal por 100g
          </strong>
        </p>
        <hr />
        <p
          class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
        >
          Mango tropical: joya dorada de dulzura veraniega. Pulpa jugosa y vibrante color en cada bocado, una experiencia exquisita. ðŸ¥­ðŸŒž
        </p>
      </div>
      <span
        class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
      />
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
